src/monster.cpp

```diff
@@ -2639,6 +2629,41 @@ void monster::add_msg_player_or_npc( const game_message_type type,
     }
 }
 
+units::mass monster::get_carried_weight()
+{
+    units::mass total_weight = 0_gram;
+    if( tack_item ) {
+        total_weight += tack_item->weight();
+    }
+    if( storage_item ) {
+        total_weight += storage_item->weight();
+    }
+    if( armor_item ) {
+        total_weight += armor_item->weight();
+    }
+    for( const item &it : inv ) {
+        total_weight += it.weight();
+    }
+    return total_weight;
+}
+
+units::volume monster::get_carried_volume()
+{
+    units::volume total_volume = 0_ml;
+    for( const item &it : inv ) {
+        total_volume += it.volume();
+    }
+    return total_volume;
+}
+
+void monster::move_special_item_to_inv( cata::optional<item> *it )
```

Comment:
```suggestion
void monster::move_special_item_to_inv( cata::optional<item> &it )
```

Image what happens when somebody calls this with a null pointer. Using a reference here prevents that.